### DISCUSSION TITLE
Support mcap files with schemaless channels containing json messages

### DISCUSSION THREAD

**Discussion by defunctzombie** [foxglove/community#443] [Q&A] [answered]
Studio is a great tool for quickly plotting or looking at data. It could be cool to support opening mcap files where the message data contains JSON but the user did not bother to specify a schema (i.e. schema id `0` indicating the channel is schemaless). Its not uncommon to avoid trying to specify a schema to test our writing some json data to an mcap file.

[output.mcap.zip](https://github.com/foxglove/studio/files/9568866/output.mcap.zip)

Related: A basic schema of `{ type: object, properties: {}}` does work tho isn't obvious if you are new to json schema that this might work.

**Comment by therishidesai** [comment]
There may be some over lap with the feature I want: [https://github.com/foxglove/community/issues/330](https://github.com/foxglove/community/issues/330)

**Comment by amacneil** [comment, marked as answer]
This was added in https://github.com/foxglove/studio/pull/5351

MCAP files should set `schema_id=0` for channels with no schema, per the [spec](https://mcap.dev/specification/index.html#channel-op0x04).
